### PR TITLE
Allow boolean subschemas everywhere.

### PR DESCRIPTION
**Pull request by handrews**
This implements issue #101

### CODE REVIEW

**Comment by handrews** [comment]
Added a commit that puts JSON Reference in the meta-schema.  This addition does not change any specified behavior at all, it just further brings the meta-schema in line with the latest draft.  I added it here because it was easier than having to merge two different subschema fixes.

**Comment by handrews** [comment]
@awwright @Relequestual any concerns holding this up?

**Comment by epoberezkin** [comment]
lgtm

as a side comment - we are on the right path of differentiating schemas and subschemas :)

File: hyper-schema.json [hunk 1]
```diff
@@ -69,9 +59,17 @@
         }
     },
     "definitions": {
+        "subSchema": {
+            "anyOf": [
+                {"$ref": "#"},
+                {"type": "boolean"},
+                {"$ref": "http://json-schema.org/draft/hyper-schema#/definitions/jsonReference"}
```
Comment (epoberezkin) [inline hyper-schema.json]: I think it should be http://json-schema.org/draft/schema#/definitions/jsonReference

Comment (handrews) [inline hyper-schema.json]: Good catch, thanks!  Fixed.

File: schema.json [hunk 2]
```diff
@@ -3,10 +3,24 @@
     "$schema": "http://json-schema.org/draft/schema#",
     "description": "Core schema meta-schema",
     "definitions": {
+        "jsonReference": {
```
Comment (CroniD) [inline schema.json]: Shouldn't the property "$ref" be required?

Comment (handrews) [inline schema.json]: Yeah, I'll fix that, thanks!

Comment (handrews) [inline schema.json]: Fixed.

**Comment by handrews** [comment]
@awwright @Relequestual @epoberezkin 

I have taken the meta-schema changes out of this and will re-add once PR #168 is accepted, at which point the changes will be simpler.

I will do the same with PR #167 (the version of this that includes root schemas), which I definitely prefer to this version now.  So if there's any consensus on that please comment so I can close this version.

**Comment by handrews** [comment]
@awwright if you're pretty sure that #167 is the better approach (and at this point I prefer it) I can close this one out.

File: jsonschema-core.xml [hunk 3]
```diff
@@ -189,7 +189,7 @@
                 <t>
                     A JSON Schema document, or simply a schema, is a JSON document used to describe an instance.
                     A schema is itself interpreted as an instance.
-                    A JSON schema MUST be an object.
+                    A JSON Schema document MUST consist of an object, known as the root schema.
                 </t>
```
Comment (epoberezkin) [inline jsonschema-core.xml]: This language is not correct. "Consists of" means "has parts". Previous "must be" is correct.

File: jsonschema-validation.xml [hunk 4]
```diff
@@ -535,12 +532,11 @@
                 </t>
                 <t>
                     This keyword's value MUST be an object. Each property specifies a dependency.
-                    Each dependency value MUST be an object or an array.
+                    Each dependency value MUST be an array or a valid subschema.
                 </t>
```
Comment (epoberezkin) [inline jsonschema-validation.xml]: I would add "array of strings"?

File: jsonschema-core.xml (hunk 3 above)
Comment (handrews) [inline jsonschema-core.xml]: I think it's fixed in #167, which currently has more support than this PR.  If that changes I will backport @awwright's language fixes to this PR.

**Comment by awwright** [comment]
#167 was used instead.

**Pull request closed without merging**